### PR TITLE
[16.01] Add admin documentation for framework dependencies (wheels)

### PR DESCRIPTION
**Pull request by natefoo**
We should probably highlight the **Galaxy job handlers** section in the release notes since some Galaxy servers will need to do this for job handling to continue working correctly.

As I got deeper into writing this I lost focus a bit - if there are things that I should cover and I didn't, please let me know. Likewise, if I went too into depth on anything, please point it out.

### CODE REVIEW

File: doc/source/admin/framework_dependencies.rst [hunk 1]
```diff
@@ -0,0 +1,573 @@
...
+    # To activate this environment, use:
+    # $ source activate galaxy
+    #
+    # To deactivate this environment, use:
+    # $ source deactivate
+    #
+    $ source activate galaxy
+    discarding /home/nate/conda/bin from PATH
+    prepending /home/nate/conda/envs/galaxy/bin to PATH
+    $ pip install --index-url=https://wheels.galaxyproject.org/simple/ -r lib/galaxy/dependencies/requirements.txt
```
Comment (jmchilton) [inline doc/source/admin/framework_dependencies.rst:389]: We never merged your conda PR - I had open questions. Should I just merge despite my concerns or do you want to address them?

File: doc/source/admin/framework_dependencies.rst [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+external to the Galaxy source tree, which can be done either by not using
+``run.sh`` directly (an example of this can be found under the `Supervisor`_
+section) or using the ``--no-create-venv`` option, explained in the `Options`_
+section. It is also possible to force Galaxy to start without a virtualenv at
+all, but you should not do this unless you know what you're doing.
+
+To manually create a virtualenv, you will first need to obtain virtualenv.
+There are a variety of ways to do this:
+
+- ``pip install virtualenv``
```
Comment (martenson) [inline doc/source/admin/framework_dependencies.rst:68]: is it worth adding `brew install virtualenv` ?

File: doc/source/admin/framework_dependencies.rst (hunk 1 above)
Comment (natefoo) [inline doc/source/admin/framework_dependencies.rst:389]: No I'm going to do them. My queue this week has been getting longer and the priority of other things keeps getting higher.

File: doc/source/admin/framework_dependencies.rst [hunk 3]
```diff
@@ -0,0 +1,573 @@
...
+  ``virtualenv.py`` script contained within as ``python virtualenv.py
+  /path/to/galaxy/virtualenv``
+
+Once this is done, create a virtualenv. In our example, the virtualenv will
+live in ``/srv/galaxy/venv`` and the Galaxy source code has been cloned to
+``/srv/galaxy/server``.
+
+.. code-block:: console
+    
+    $ virtualenv /srv/galaxy/venv
```
Comment (martenson) [inline doc/source/admin/framework_dependencies.rst:83]: the text above operates with `.venv` not `venv`

File: doc/source/admin/framework_dependencies.rst (hunk 1 above)
Comment (natefoo) [inline doc/source/admin/framework_dependencies.rst:389]: But thanks for the reminder, the priority of that task has just increased. =)

File: doc/source/admin/framework_dependencies.rst [hunk 4]
```diff
@@ -0,0 +1,573 @@
...
+If using `Pulsar`_'s option to set metadata on the remote server, the same
+conditions as with `Galaxy job handlers`_ apply. You should create a virtualenv
+on the remote resource, install Galaxy's dependencies in to it, and set an
+``<env>`` tag pointing to the virtualenv's ``activate`` as in the `Galaxy job
+handlers`_ section. Instructions on how to create a virtualenv can be found
+under the `Managing dependencies manually`_ section.
+
+.. _Pulsar: http://pulsar.readthedocs.org/
+
+Conda
```
Comment (martenson) [inline doc/source/admin/framework_dependencies.rst:235]: As I understand it Conda is very much experimental feature and thus I feel it belongs to the very end of this document.

Comment (natefoo) [inline doc/source/admin/framework_dependencies.rst:235]: Conda for dependencies, yes. However, we have people who use Conda's python as the Python used to start/run Galaxy, so this documentation is relevant to that scenario.

File: doc/source/admin/framework_dependencies.rst (hunk 3 above)
Comment (natefoo) [inline doc/source/admin/framework_dependencies.rst:83]: The point of this section is to show creating a virtualenv in an alternate, admin-controlled location.

File: doc/source/admin/framework_dependencies.rst (hunk 2 above)
Comment (natefoo) [inline doc/source/admin/framework_dependencies.rst:68]: Done.

**Comment by martenson** [comment]
:+1:  very nice docs

**Comment by nsoranzo** [comment]
:+1: Thanks @natefoo!

**Pull request merged**